src/server/frontend/wayland/wl_surface.cpp

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright © 2018 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: Christopher James Halse Rogers <christopher.halse.rogers@canonical.com>
+ *              William Wold <william.wold@canonical.com>
+ */
+
+#include "wl_surface.h"
+
+#include "core_generated_interfaces.h"
+#include "wl_mir_window.h"
+#include "wayland_connector.h"
```

Comment:
The only things used from wayland_connector.h are session_for_client() and run_unless().
Surely there's a better place to put them so we can lose this dependency?

Reply:
See above ^^, There's not a better place, but I can make one.